### DISCUSSION TITLE
Site affinity in distributed setup

### DISCUSSION THREAD

**Discussion by hoalex** [Security-Onion-Solutions/securityonion#8695] [Unsupported Versions]
Hi everyone,

we are considering deploying SO according to the distributed architecture model described in the documentation, with a total of 2 forward nodes, 1 manager node and 2 search nodes.
We have two sites/datacenters, one primary and one for backup purposes -- the idea is to have one component run on each site, with the manager node always residing at the primary site, like this:

```
  primary site                  standby site

┌──────────────────────────┐   ┌───────────────────────┐
│                          │   │                       │
│                          │   │                       │
│       ┌───────────────┐  │   │   ┌────────────────┐  │
│       │               │  │   │   │                │  │
│       │ forward node  │  │   │   │  forward node  │  │
│       │               │  │   │   │                │  │
│       └───────┬───────┘  │   │   └───────┬────────┘  │
│               │          │   │           │           │
│               │          │   │           │           │
│       ┌───────▼──────┐   │   │           │           │
│       │              │   │   │           │           │
│       │ manager node ◄───┼───┼───────────┘           │
│       │              │   │   │                       │
│       └───────┬──────┴───┼───┼────────────┐          │
│               │          │   │            │          │
│       ┌───────▼──────┐   │   │   ┌────────▼───────┐  │
│       │              │   │   │   │                │  │
│       │ search node  │   │   │   │ search node    │  │
│       │              │   │   │   │                │  │
│       └──────────────┘   │   │   └────────────────┘  │
│                          │   │                       │
│                          │   │                       │
└──────────────────────────┘   └───────────────────────┘
```

So far, so good -- however, one aspect I don't like is that the collected data that is fed through Logstash and Redis on the manager node ends up being distributed between the Elasticsearch instances on the 2 search nodes, meaning that some logs of the primary site may end up on the ES instance on the backup site and vice versa.
Ideally, I would like to keep these separate (what I call "site affinity"), meaning that data that originated from the forward node on the primary site is exclusively written to the ES instance on the primary site, and vice versa for the backup site.

I know that one could setup a true ES cluster on the search nodes, but this would mean that I need double the storage on both sites, in order to host not only the primary but also the replica shards. I simply want to have the data separated between the sites.

Does anyone know of a way to accomplish this?

**Comment by reyesj2** [comment]
To separate the data then I think you would need two security onion instances 1 manager 1 fwd node 1 search node for both datacenters. That would ensure data is not mixed. 

If just keeping a separation of events by datacenter is what you're looking for you can simply filter by hostname of sensor. That way you maintain your centralized log management and don't have to manage two Security Onion instances.

**Reply by hoalex** [reply]
I thought about using two manager nodes as well, so effectively having two independent SO grids. What I don't like about this approach is that all configuration on the grid needs to be manually synchronized between the two sites. Let's say I want to customize the Wazuh ruleset, or open ports via so-allow -- everything needs to be done on two sites individually, as the goal is to keep the config the same :-(

If it would only be a matter of rsync'ing a directory or two, this would not be a huge deal, but I'm afraid it would be a lot more involved than that.

**Comment by presianbg** [comment]
Hi,

Just use heavy nodes and your problem would be solved!
![image](https://user-images.githubusercontent.com/2163486/189218597-01620ec3-f67a-4f99-a037-4617273b4f15.png)
The question is how you get the data from the `stand-by` site, when `primary` site goes down ?
